pkg/storegateway/indexheader/stream_binary_reader.go

```diff
@@ -148,7 +175,107 @@ func newFileStreamBinaryReader(path string, postingOffsetsInMemSampling int, log
 		return nil, err
 	}
 
-	return r, nil
+	return r, err
+}
+
+// loadSparseFromDisk loads in sparse index-header from disk.
+func (r *StreamBinaryReader) loadSparseFromDisk(ctx context.Context, logger log.Logger, sparseHeadersPath string, sparseData []byte, postingOffsetsInMemSampling int) (err error) {
+	span, _ := tracing.StartSpan(ctx, "index_header_load_sparse_header")
+	defer span.Finish()
+
+	start := time.Now()
+	defer level.Info(logger).Log("msg", "loaded sparse index-header from disk", "path", sparseHeadersPath, "elapsed", time.Since(start))
```

Comment:
1. I think it would be more useful to log the block ID here so that we can compare the time taken to load a sparse index-header and the time taken the load the full thing for a particular block
2. Can you double check that `time.Since(start)` is evaluated when the defer runs and not immediately? This is a golang thing that I can't remember offhand.

Reply:
2. double decked, `time.Since(start)` should be evaluated when defer runs

Reply:
I don't think that's correct. Take a look at these two playground snippets:

[Bare defer](https://go.dev/play/p/BIq0miFfwG3)
[Wrapped in a closure](https://go.dev/play/p/fzPNyWWKpoZ)